content/v0.0.1/index.md

```diff
@@ -6,3 +6,290 @@ aliases: ["/en/"]
 # Conventional Logs v0.0.1
 
 ## Summary
+
+The conventional logs specification is an opinionated convention
+aiming to set an industry standard to ensure search-ability
+and enhance security in logs.
+
+Each log entry could be encoded in different formats, `json`, `xml`, or even a custom one.
+Most of the examples in this page will be written as `json`.
+
+A basic log entry should be structured as follows:
+
+---
+
+```
+{
+  "severity": "DEBUG",
+  "scope": "web-app",
+  "message": "could not save the user",
+  "timestamp": "2022-05-14T14:16:15+00:00"
+}
+```
+
+---
+
+A minimal log entry is composed by the following fields,
+and needs to be descriptive enough to determine the context related to event occurrence.
+
+1. The `severity`, also referred as `level`, describes the importance of a log entry.
+   Starting from the lower priority to the higher one, the recommended levels are ***debug***, ***info***,
+   ***warning*** (or ***warn***), ***error***, ***fatal***.
+3. Other values as `severity` can be used, but it is recommended to stick to the suggested one.
+4. The `scope` field represents the issuer of the log entry.
+   This is useful especially in contexts where a log entry may be created from multiple services.
+5. In complex systems the `scope` could also point to the node/service that caused the log entry.
+6. The `message` is the core information of a log entry, and describes an event that occurred in a system.
+7. When the same event happens multiple times, the `message` must always be the same and other fields must be used as
+   discriminant.
+8. The `timestamp` field determines when a log entry has been created and contains date+time, it is optional to pass
+   timezone information as well.
+10. Each of fields can be renamed and adapted to a company need. An example: `message` may be reported as `event`
+    or `msg` etc.
+11. The conventional log specification encourages adding more custom fields to better describe the log entry.
+12. Sensible data, such as customer information ([PII](https://en.wikipedia.org/wiki/Personal_data)),
+    auth token or any other secret must not be added to the entry.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and
+“OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+When the same event happens multiple times, the message must always be the same and other fields must be used as
+discriminant.
+
+1. A field for the `message` MUST be provided in each log entry.
+2. The content of a `message` field MUST be a description of an event occurrence.
+3. The content of the `message` field SHOULD be unique within your system.
+4. The content of the `message` field MUST NOT contain variable content and MUST be a static.
+5. The `message` field MAY be named differently.
```

Comment:
This breaks the open-closed principle. The standard should define a set of must-have fields, and yet provide the means to extend it without modifying the base.

Reply:
This is needed because different platforms consume and interact with logs in different ways (GCP does that differently than DD, and so on).

So, IMHO, it is ok to have a `message` field named in a different way, as long as it is explicit that the "message" concept is part of a log.

Maybe I should rephrase it in a way that explains that better?
What are your 2 cents?

Reply:
Still, mutable fields = no data contract = no standard. Ideally any non-matching interfaces should use adapters to send/receive data from standard-compliant endpoints.

Reply:
Let's see how we can make this happen.
I do hope this won't be a blocker for the convention, I'd need to spend some time investigating if it is possible to customize platforms that store logs so that we can unsure the standard can grow